lib/suseconnect.pm

```diff
@@ -0,0 +1,107 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Register base product & extensions with SUSEConnect
+# Maintainer: Jiawei Sun <jwsun@suse.com>
+
+package suseconnect;
+
+use base Exporter;
+use Exporter;
+
+use strict;
+use testapi;
+
+
+our @EXPORT = qw (suseconnect_registeration command_register);
+
+sub suseconnect_registeration {
+    my $product_version = get_var('VERSION');
+
+    # register base prodcut
+    command_register($product_version);
+    my @scc_addons = split(/,/, get_var('SCC_ADDONS', ''));
+
+    # register extensions without reg code
+    for my $addon (@scc_addons) {
+        next unless grep { $addon eq $_ } qw(asmm contm lgm pcm tcm wsm);
+        command_register($product_version, $addon);
+    }
+    assert_script_run 'zypper lr';
+}
+
+sub command_register {
+    my ($version, $addon, $addon_regcode) = @_;
+    my $arch = get_var("ARCH");
```

Comment:
Also here, unsafe. You should either provide a sane default or use `get_required_var`